ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -1503,6 +1503,11 @@ impl<'a> Assemble<'a> {
             "llvm.assume" => Ok(()),
             "llvm.lifetime.start.p0" => Ok(()),
             "llvm.lifetime.end.p0" => Ok(()),
+            x if x.starts_with("llvm.smax") => {
```

Comment:
Do we know for sure that `llvm.smax` is never inlined at the MIR level?

Reply:
I see no evidence of that, but really it's beyond my purview.

Reply:
The problem is, if an intrinsic is inlined at the codegen level, then we trace the inlined version and then call it a second time.

That's probably OK (if a little iffy) for an operation like smax, but you certainly don't want to do that for operations that are expensive (in terms of performance, since you would incur the cost twice) or have side effects (which would break program semantics).

I'd like to hear what @ptersilie thinks, but my gut feeling is to not merge this until we have a better way of knowing whether an intrinsic has been inlined or not.